### PR TITLE
[Form][Type Date/Time] added choice_translation_domain option.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15222 and #15298 |
| License | MIT |
- [x] add test

### CODE REVIEW

**Comment by MacDada** [comment]
No tests?

**Comment by MacDada** [comment]
Will this be backported to 2.7?

File: src/Symfony/Component/Form/Extension/Core/Type/DateType.php [hunk 1]
```diff
@@ -200,6 +203,23 @@ public function configureOptions(OptionsResolver $resolver)
             );
         };
 
+        $choiceTranslationDomainNormalizer = function (Options $options, $choiceTranslationDomain) {
+            if (is_array($choiceTranslationDomain)) {
+                $default = false;
+
+                return array_merge(
```
Comment (Tobion) [inline src/Symfony/Component/Form/Extension/Core/Type/DateType.php]: semantically you want array_replace

**Comment by Tobion** [comment]
Related to #4833

File: src/Symfony/Component/Form/Extension/Core/Type/DateType.php (hunk 1 above)
Comment (aitboudad) [inline src/Symfony/Component/Form/Extension/Core/Type/DateType.php]: fixed

**Comment by aitboudad** [comment]
test added.

**Comment by Tobion** [comment]
:+1:  Good to merge IMO.

**Comment by fabpot** [comment]
Should it be considered as a new feature?

**Comment by Tobion** [comment]
IMO yes as it doesn't fix a real bug but only tries to reduce missing translation warnings (warnings are not bugs).

**Comment by stof** [comment]
:+1:

**Comment by Tobion** [comment]
Thank you @aitboudad.

**Pull request closed without merging**